src/extension/tags.c

```diff
@@ -778,19 +861,35 @@ static PHP_FUNCTION(datadog_appsec_track_user_login_success_event)
 
     zend_string *user_id = NULL;
     HashTable *metadata = NULL;
-    if (zend_parse_parameters(ZEND_NUM_ARGS(), "S|h", &user_id, &metadata) ==
-        FAILURE) {
+    zend_bool automated = false; // Don't document. Only internal usage
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "S|hb", &user_id, &metadata,
+            &automated) == FAILURE) {
         mlog(dd_log_warning, "Unexpected parameter combination, expected "
                              "(user_id, metadata)");
         return;
     }
+    zend_string *mode = get_DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING();
+    if (automated) {
+        if (mode && strncmp(ZSTR_VAL(mode), LSTRARG(MODE_DISABLED)) == 0) {
+            return;
+        }
 
-    if (user_id == NULL || ZSTR_LEN(user_id) == 0) {
-        mlog(dd_log_warning, "Unexpected empty user id");
-        return;
-    }
+        if (mode && strncmp(ZSTR_VAL(mode), LSTRARG(MODE_SAFE)) == 0) {
+            if (user_id != NULL && is_user_id_sensitive(user_id)) {
+                zend_string_release(user_id);
+                user_id = ZSTR_EMPTY_ALLOC();
+            }
 
-    dd_find_and_apply_verdict_for_user(user_id);
+            if (metadata != NULL && zend_array_count(metadata) > 0) {
+                zend_hash_clean(metadata);
+            }
```

Comment:
After everything has been cleaned up, if the user is empty, what is the point of continuing? There is no valuable information...